### ISSUE TITLE
Provisioning Datasource: InfluxDB

### ISSUE THREAD

**Issue by auwaerter** [grafana/helm-charts#2243]
I was using the following provisioned datasource for my InfluxDB with grafana 9.3.6 which was working fine:
```
      - name: Icinga2-InfluxDB
        type: influxdb
        access: proxy
        database: icinga2
        url: http://127.0.0.1:2003
        uid: icinga2-id
        jsonData:
          timeInterval: "15s"
```
Now i've updated to grafana version 9.4.1 with HELM Chart 6.51.1 and it seems like i can't provision the database name, as the following error is being returned with using the Test button:

![image](https://user-images.githubusercontent.com/6923945/222141879-947cd5ec-3bb1-4760-ac25-44dcd50c5105.png)

Also when i'm adding the correct name inside grafana, the same error is being returned. When i'm adding the datasource manually with the grafana UI the connection is working fine.

Did something changed within grafana 9.4.1 where the provisioning is not working any longer? Or am i using wrong values for Influx provisioning?

Thanks! :)

**Comment by auwaerter** [comment]
**Update:** HELM Chart 6.51.3 seems to fixed that issue.

**Comment by auwaerter** [comment]
Update: Also with updated HELM Chart 6.51.3 it is not working.

**Comment by Ramshield** [comment]
I got the same issue with helm chart 6.52.4 and image version 9.4.3.

**Comment by ulexxander** [comment]
Same issue, chart 6.52.7.

Using Terraform:

```hcl
resource "helm_release" "grafana" {
  name             = "grafana"
  repository       = "https://grafana.github.io/helm-charts"
  chart            = "grafana"
  version          = "6.52.7"
  namespace        = "metrics"
  create_namespace = true

  values = [
    yamlencode({
      datasources = {
        "datasources.yaml" = {
          apiVersion = 1
          datasources = [
            {
              name      = "InfluxDB - telegraf"
              type      = "influxdb"
              access    = "proxy"
              url       = "http://influxdb:8086"
              isDefault = true
              database  = "telegraf"
            }
          ]
        }
      }
    })
  ]
}
```

**Comment by ulexxander** [comment]
Noticed that if you restart Grafana (delete pod in my case, automatically recreated), then datasource database is set correctly and starts working.

So the problem may be somewhere in initial inserting logic, because subsequent restarts of Grafana only update datasources, which fixes database option.

![image](https://user-images.githubusercontent.com/55105865/230637775-c6d32175-9229-477f-bfe0-8aedbffa8b11.png)

**Comment by auwaerter** [comment]
Issue has been fixed for me with an update to v9.5.1
